### ISSUE TITLE
Todo List

### ISSUE THREAD

**Issue by Nugine** [Nugine/const-str#12]
New API:
+ [x] `compare!`
+ [x] `strip_prefix!`
+ [X] `strip_suffix!`
+ [x] `join!`
+ [x] `sorted!`

Upstream issues:
+ [x] <https://github.com/rust-lang/rust/issues/108231>

**Comment by Scripter17** [comment]
I don't suppose I could bribe you with $20 USD to do `join!`? It would help a lot in a niche personal project I'm working on

Also I assume the intent is to take an array and/or slice of string literals and a joiner string? I can't imagine it'd be anything else but I just want to be sure

**Comment by Nugine** [comment]
> I don't suppose I could bribe you with $20 USD to do `join!`? It would help a lot in a niche personal project I'm working on
> 
> Also I assume the intent is to take an array and/or slice of string literals and a joiner string? I can't imagine it'd be anything else but I just want to be sure

Yes. The intent is to take string slices and concat them using a specified separator. I'll give it a try since someone needs it.

**Comment by Scripter17** [comment]
Managed to make it partially work, but not with slices.

```Rust
macro_rules! join {
    (join=$join:expr, $x:expr) => ($x);
    (join=$join:expr, $x:expr, $($y:expr),+) => {
        concat!($x, $join, join!(join=$join, $($y),+))
    };
}
```

**Comment by Nugine** [comment]
https://docs.rs/const-str/0.5.7/const_str/macro.join.html

**Comment by Scripter17** [comment]
I currently do not have $20 to give you. I swear I'm not just going back on the promise; Just give me until April.

**Comment by Scripter17** [comment]
Finally have the money. Do you have a paypal account?

**Comment by Nugine** [comment]
Closing this issue in favor of #27
